### PR TITLE
SALTO-1817-jira-deploy-e2e

### PR DESCRIPTION
**Pull request by alonstern**
Added e2e for Jira deploy.

---
_Release Notes_: 
None

---
_User Notifications_: 
None

### CODE REVIEW

File: packages/jira-adapter/src/reference_mapping.ts [hunk 1]
```diff
@@ -131,7 +131,7 @@ export const referencesRules: referenceUtils.FieldReferenceDefinition<never>[] =
     target: { type: 'Project' },
   },
   {
-    src: { field: 'id', parentTypes: ['ProjectRole'] },
```
Comment (alonstern) [inline packages/jira-adapter/src/reference_mapping.ts:134]: 😳

File: packages/jira-adapter/e2e_test/instances/index.ts [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+  const issueTypeScreenScheme = new InstanceElement(
+    randomString,
+    findType('IssueTypeScreenScheme', fetchedElements),
+    createIssueTypeScreenSchemeValues(randomString, fetchedElements),
+  )
+
+  const fieldConfigurationScheme = new InstanceElement(
+    randomString,
+    findType('FieldConfigurationScheme', fetchedElements),
+    {
```
Comment (noamhammer) [inline packages/jira-adapter/e2e_test/instances/index.ts:99]: nit: why you didn't add value creator for this type as well? (same question for issueTypeScheme as well)

**Review by noamhammer — APPROVED** [review]

File: packages/jira-adapter/src/reference_mapping.ts (hunk 1 above)
Comment (alonstern) [reply packages/jira-adapter/src/reference_mapping.ts:134]: Yea, when trying to deploy something with a reference to project role

File: packages/jira-adapter/e2e_test/utils.ts [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+import { InstanceElement, isInstanceElement, isObjectType, ObjectType, Element, ElemID, ReferenceExpression } from '@salto-io/adapter-api'
+
+
+export const findType = (name: string, allElements: Element[]): ObjectType => {
```
[the remark this answers is not in the document]
Comment (alonstern) [reply packages/jira-adapter/e2e_test/utils.ts:19]: when `findObjectType` does not find the type it returns undefined and does not throw an exception. But I will change this function to use `findObjectType`

File: packages/jira-adapter/e2e_test/utils.ts [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+  const type = allElements.filter(isObjectType).find(e => e.elemID.name === name)
+  if (type === undefined) {
+    throw new Error(`${name} type was not found in received elements`)
+  }
+  return type
+}
+
+export const findInstance = (id: ElemID, allElements: Element[]): InstanceElement => {
+  const instance = allElements.filter(isInstanceElement).find(e => e.elemID.isEqual(id))
```
[the remark this answers is not in the document]
Comment (alonstern) [reply packages/jira-adapter/e2e_test/utils.ts:28]: The `filter` is to avoid having to cast instance to InstanceElement

**Pull request merged**